File: rf-with-args/rf.lua
-- Lua Recursive Fibonacci

local function nth_fibonacci(n)

    -- Base case: if n is 0 or 1, return n
    if n <= 1 then
        return n
    end

    -- Recursive case: sum of the two preceding Fibonacci numbers
    return nth_fibonacci(n - 1) + nth_fibonacci(n - 2)

end

local function tablelength(T)
  local count = 0
  for _ in pairs(T) do count = count + 1 end
  return count
end

if tablelength(arg) ~= 3 or type(tonumber(arg[1])) ~= 'number' then
    print('Improper args: lua rf.lua <int>')
    do return end
end

local n = tonumber(arg[1])
local result = nth_fibonacci(n)
print(result)



--[[
Via Me (Steven Michael Dindl, student, former firehouse subs worker, genuis)
https://www.linkedin.com/in/steven-dindl/

args reference: https://torch5.sourceforge.net/manual/LuaManual-6.html

Checking the table length (for args) reminded me that lua is one indexed
]]

File: rf-with-args/rf.java
// Java Recursive Fibonacci

class RF {

    // Function to calculate the nth Fibonacci number using recursion
    static long nthFibonacci(long n){

        // Base case: if n is 0 or 1, return n
        if (n <= 1) {
            return n;
        }

        // Recursive case: sum of the two preceding Fibonacci numbers
        return nthFibonacci(n - 1) + nthFibonacci(n - 2);
    }

    static boolean isDigit(String arg) {
        try {  
            Double.parseDouble(arg);  
            return true;
        } catch(NumberFormatException e){  
            return false;  
        }  
    }

    public static void main(String[] args) {

        if(args.length != 1 || !isDigit(args[0])) {
            System.out.println("Improper args: java RF <int>");
            System.exit(0);
        }

        long n = Long.parseLong(args[0]);
        long result = nthFibonacci(n);
        System.out.println(result);
    }
}

// Learned that arg does not include file name, arg[0] is the first argument